docs/rules/no-unused-expressions.md

```diff
@@ -39,6 +39,21 @@ a && b()
 a, b()
 
 c = a, b;
+
+a() && function namedFunctionInExpressionContext () {f();}
+
+(function anIncompleteIIFE () {});
+
```

Comment:
I've amended the commit to reiterate the directive information in the beginning of Rule Details (and to fix a grammar nit).